### DISCUSSION TITLE
How to decide an ideal step-size for PAG reconstruction in martensitic steel?

### DISCUSSION THREAD

**Discussion by Deep0161** [ORTools4MTEX/ORTools#29] [General]
First of all thank you for the reconstruction algorithm. I would like to know, is there a possibility of determining an ideal scan step-size for my EBSD scan. Which will result in a parent grain reconstruction with high confidence? Like is there any possibility where the step-size can be related to Lath width or any other microstructural features?

**Comment by frankNiessen** [comment]
Dear Deep0161, 

If the only aim is to obtain the parent grain microstructure then you can work with relatively low spatial resolution. As long as you capture multiple characteristic orientations per parent grain, the reconstruction should be solid. This might be at the expense of details like annealing twins, which even in a high-resolution map are not straight forward to capture. 
If you want to also conduct variant analysis, the story is a bit different. Here you of course need to capture sufficient detail within the martensitic structure in order to get a representative dataset. 
If you look at the lath martensite map of [Example 1](https://github.com/ORTools4MTEX/ORTools#example-7-23-mar-2022) you will be able to see that the map resolution is quite low and that there is a significant fraction of unindexed points, still the parent grain reconstruction works well. So in the end it is a trade-off between parent microstructure statistics vs. child microstructure statistics vs. resource allocation. 

I hope this is helpful to you,
Frank

**Reply by Deep0161** [reply]
Thank you very much. 
I would like to know about available options for twin boundary demarcation? Is there a way, I have found in literatures people doing it on the basis of CSL 3 Boundaries. However, I have found they don't always predict the twins properly in reconstructed parent even though these kind of boundaries are a part of twins in FCC.
What are your takes on it?

**Reply by frankNiessen** [reply]
Yes, we usually consider the CSL3 boundaries, which you can do as shown below. Do you have any more information on that? I think the annealing twins are associated with a CSL3 boundary, and when you see the boundary moved by small distances this usually has to do with the difficulty of reconstructing annealing twins properly, all current methods for parent grain reconstruction are challenged by this use case.

```gB = job.grains(job.csParent,job.csParent)
gP = unique(gB.grainId,'rows');
gP3 = gP(angle(inv(parentGrains(gP(:,1)).meanOrientation).* parentGrains(gP(:,2)).meanOrientation,CSL(3,job.p2c.CS)) < 5*degree,:);
gB3 = gB.selectByGrainId(gP3);

hold on
plot(gB,'linewidth',1.5)
hold on
plot(gB3,'lineColor','cyan','linewidth',1.5,'DisplayName','CSL 3')```

**Reply by Deep0161** [reply]
Yes, Frank. I am also using the same logic. However, for my reconstructed dataset this doesn't always give a satisfactory result even with a 5degree threshold (Pl. see the attached image done with CSL 3
![newfile](https://user-images.githubusercontent.com/53735369/215506576-1c52e739-8adc-401e-a9e0-cde919ab3e65.jpg)
) In places where it looks like apparent twins, CSL can't capture the same. However, when I do it with a higher threshold it captures.

**Reply by frankNiessen** [reply]
I can see several twins that seem to be correlating with the marked CSL3 boundary, but also some that do not show the CSL3 relationship. Which threshold do you say works? Does that threshold then also mark twin boundaries that definitely are not such?

**Reply by Deep0161** [reply]
A threshold of 8 degrees, however, also marks certain grain boundaries which are not actually twins. There is a trade-off.

**Reply by frankNiessen** [reply]
This does not sound entirely unreasonable to me. In the case of reconstructed austenite maps the quality of the reconstruction is also critical for the possibility of identifying annealing twins. If you for example determine an average orientation relationship that is few degrees away from your actual orientation relationship, then you will also calculate austenite orientations that are misoriented from their real prior austenite orientations. This can easily give few more degrees. Maybe you could also check what the actual misorientation at the boundaries that are obviously associated with a twin are and see how these relate to Sigma 3 boundaries.

**Reply by Deep0161** [reply]
Thnak you for the clarification. Yes doing a misorientation plot helps to some extent in getting a ballpark.
Thank you once again.
